nncf/quantization/layers.py

```diff
@@ -219,6 +226,13 @@ def enable_quantizers(self):
         self.originally_disabled = []
 
 
+class HookAutoRemove:
+    def __init__(self, hook):
+        self.hook = hook
+    def close(self):
```

Comment:
AFAIK, `close` should work context manager only. It seems you to override `__del__` method

Reply:
You are right. I 've made blind copy from LoadStateListener class.
Thanks a lot!